Split Gaffer container into separate containers

cybermaggedon opened the pull request:
Here's an update which splits the Gaffer container into separate containers: Hadoop, Accumulo, Zookeeper, Wildfly.

There are several advantages to this approach:

- Operator has the option of exposing management ports on any of the services.
- Lifecycle management, individual components can be upgraded separately.
- Failure of an individual component can be detected, and restart is more local in scope, which helps service uptime.

This has changes requested from p013570 in a previous pull request.

cybermaggedon commented:
Docker repo names and README adjusted for 'gchq' repo.

t616178 commented on Dockerfile.accumulo:
```diff
@@ -0,0 +1,20 @@
+# limitations under the License.
+##########################################################
+
+FROM cybermaggedon/accumulo:1.8.0
```
There are currently some unresolved issues with Accumulo 1.8.0 and Gaffer (as well as a general lack of testing) - would it be possible to use Accumulo 1.7.2 for the moment?

cybermaggedon replied:
Yep, no problem.

t616178 commented on README.md:
````diff
@@ -1,57 +1,106 @@
+Wildfly needs no persistent state.
 
-  docker run -it -p 8080:8080 -v /data/gaffer:/data gchq/gaffer:0.4.4
+```
````
Would it be possible (or useful) to have a shell script that can launch everything in one go?

cybermaggedon commented:
- Accumulo is at 1.7.2.
- Added a run_gaffer shell script.
- Added some Kubernetes templates. Snippets from what I'm using to run Gaffer in Google cloud.
